### PR TITLE
feat: support multiple contracts in Documents Batch Transition

### PR DESCRIPTION
**Pull request by shumkov**

### CODE REVIEW

File: lib/errors/MissingDocumentTransitionActionError.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const ConsensusError = require('./ConsensusError');
+
+class MissingDocumentTransitionActionError extends ConsensusError {
+  constructor(rawDocumentTransition) {
```
Comment (shuplenkov) [inline lib/errors/MissingDocumentTransitionActionError.js:4]: JS doc missing

File: lib/document/stateTransition/validation/data/fetchDocumentsFactory.js [hunk 2]
```diff
@@ -5,44 +5,43 @@
...
    * @return {Document[]}
    */
-  async function fetchDocuments(dataContractId, documentTransitions) {
-    // Group Document IDs by types
-    const documentIdsByTypes = documentTransitions.reduce((obj, document) => {
-      if (!obj[document.getType()]) {
+  async function fetchDocuments(documentTransitions) {
+    // Group document transitions by contracts and types
+    const transitionsByContractsAndTypes = documentTransitions.reduce((obj, dt) => {
+      const uniqueKey = `${dt.getDataContractId()}${dt.getType()}`;
```
Comment (shuplenkov) [inline lib/document/stateTransition/validation/data/fetchDocumentsFactory.js:16]: Maybe it's better to have a method for this?

Comment (antouhou) [reply lib/document/stateTransition/validation/data/fetchDocumentsFactory.js:16]: You mean like `document.getUniqueKey()`?

Comment (shumkov) [reply lib/document/stateTransition/validation/data/fetchDocumentsFactory.js:16]: Not sure it makes sense since we need it only in one place - here.

Comment (shumkov) [reply lib/document/stateTransition/validation/data/fetchDocumentsFactory.js:16]: @jawid-h  what do you think?

Comment (jawid-h) [reply lib/document/stateTransition/validation/data/fetchDocumentsFactory.js:16]: Just a local identifier, so I think it is fine

**Pull request merged**